src/hotspot/share/prims/jvmtiEventController.cpp

```diff
@@ -367,13 +367,9 @@ void JvmtiEventControllerPrivate::enter_interp_only_mode(JvmtiThreadState *state
     return;  // EnterInterpOnlyModeClosure will be executed right after mount.
   }
   EnterInterpOnlyModeClosure hs;
-  if (target->is_handshake_safe_for(current)) {
-    hs.do_thread(target);
-  } else {
-    assert(state->get_thread() != NULL, "sanity check");
-    Handshake::execute(&hs, target);
-    guarantee(hs.completed(), "Handshake failed: Target thread is not alive?");
-  }
+  assert(state->get_thread() != NULL, "sanity check");
```

Comment:
Existing: We have already performed this check. We set `target` above and returned if it was `NULL`.

Reply:
We also no longer need L358 as `current` is now unused.

Reply:
Removed unused variable, moved the `state != NULL` assert up and removed the `state->get_thread() != NULL` assert as it's unnecessary.